tests/transforms/test_transform_can_be_hoisted.py

```diff
@@ -0,0 +1,94 @@
+from xdsl.builder import Builder, ImplicitBuilder
+from xdsl.dialects import builtin, func, scf, test
+from xdsl.dialects.arith import Addf
+from xdsl.dialects.builtin import (
+    FunctionType,
+    IndexType,
+    ModuleOp,
+    f32,
+    i32,
+)
+from xdsl.dialects.scf import For, Yield
+from xdsl.ir import Block, BlockArgument, Region
+from xdsl.transforms.loop_invariant_code_motion import can_be_hoisted
+
+index = IndexType()
+
+
+def test_for_with_loop_invariant_verify():
+    """Test for with loop-variant variables"""
+
+    lower = test.TestOp(result_types=[i32])
+    upper = test.TestOp(result_types=[i32])
+    step = test.TestOp(result_types=[i32])
+    carried = test.TestOp(result_types=[i32])
+
+    a = test.TestOp(result_types=[i32])
+    b = test.TestOp(result_types=[i32])
+    block = Block(
+        arg_types=(
+            index,
+            index,
+        )
+    )
+    region = Region(block)
+    with ImplicitBuilder(block) as (arg0, _):
+        op1 = test.TestOp(result_types=[i32])
+        op2 = test.TestOp(result_types=[i32])
+        op3 = test.TestOp([op1, op2, arg0], result_types=[i32])
+        f = Yield(op3)
+    for_op = For(lower, upper, step, [carried], region)
+    block0 = Block([a, b, for_op])
+    _region_outer = Region(block0)
+
+    assert can_be_hoisted(op1, region)
+    assert can_be_hoisted(op2, region)
+    assert can_be_hoisted(op3, region)
```

Comment:
I'm not sure what exactly is wrong, but to my understanding this assertion should fail; `op3` is using `op1` and `op2` which are inside the loop at this point, so `op3` should *not* be considered hoistable.

The point is to check if it is immediately hoistable, and hoist them one by one; i.e., `op3` would be hoistable if `op1` and `op2` were hoisted *before* checking it.

Reply:
Just noticed my unclear selection of lines; I'm referring to `can_be_hoisted(op3, region)` specifically; `op1` and `op2` seem right to me

Reply:
You are correct, I have modified the logic and added explicit testcase that Sasha was mentioning.

Reply:
This now seems sensible to me, thanks!